### PR TITLE
[BugFix] Fix potential bug: bthread::execution_queue_join() on invalid id

### PR DESCRIPTION
**Pull request by sduzh**
If lake::AsyncDeltaWriter::open() failed, lake::AsyncDeltaWriter::close() may call bthread_execution_queue_join() on an invalid queue id.

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/storage/lake/async_delta_writer.cpp [hunk 1]
```diff
@@ -194,7 +197,7 @@ inline void AsyncDeltaWriterImpl::finish(Callback cb) {
 
 inline void AsyncDeltaWriterImpl::close() {
     std::lock_guard l(_mtx);
-    if (_opened && !_closed) {
+    if (_queue_id != kInvalidQueueId) {
```
Comment (imay) [inline be/src/storage/lake/async_delta_writer.cpp:200]: If you change code like this, is _closed still useful

Comment (sduzh) [reply be/src/storage/lake/async_delta_writer.cpp:200]: `_closed` is still useful for preventing re`open()` the AsyncDeltaWriter after `close()`ed:
```
inline Status AsyncDeltaWriterImpl::open() {
    std::lock_guard l(_mtx);
    if (_closed) { // <----- here
        return Status::InternalError("AsyncDeltaWriter has been closed");
    }
    if (_opened) {
        return _status;
    }
    _status = do_open();
    _opened = true;
    return _status;
}
````

**Review by wyb — DISMISSED** [review]

**Review by chaoyli — DISMISSED** [review]

**Review by imay — APPROVED** [review]

**Comment by sduzh** [comment]
run all

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**